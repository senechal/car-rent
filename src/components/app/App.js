import {
  BrowserRouter as Router,
  Switch,
  Route,
} from 'react-router-dom';
import { Layout, Icons } from '@reusable';
import Details from '@components/details';
import Search from '@components/search';


const { CenterLayout, Header, Footer } = Layout;
const { CarLogo } = Icons

const App = () => {
  return (
    <Router>
      <Layout>
        <Header><span>Rent-a-car</span> <CarLogo /></Header>
        <CenterLayout>
          <Switch>
            <Route path="/details/:vehId">
              <Details />
            </Route>
            <Route path="/">
              <Search />
            </Route>
          </Switch>
        </CenterLayout>
        <Footer>Footer</Footer>
      </Layout>
    </Router>
  )
}

export default App;
