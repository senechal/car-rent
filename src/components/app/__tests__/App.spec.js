import { render } from '@testing-library/react';
import App from '../App';

jest.mock('@components/details', () => () => 'Details');
jest.mock('@components/search', () => () => 'Search');

describe('Testing App', () => {
  it('should render correctly', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
