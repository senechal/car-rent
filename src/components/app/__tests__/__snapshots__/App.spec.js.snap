// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App should render correctly 1`] = `
<div>
  <section
    class="rac-layout"
  >
    <header
      class="rac-layout-header"
    >
      <span>
        Rent-a-car
      </span>
       
      <span
        class="anticon rac-car-logo"
        role="img"
      >
        <svg
          aria-hidden="true"
          class=""
          fill="currentColor"
          focusable="false"
          height="1em"
          width="1em"
        >
          car.svg
        </svg>
      </span>
    </header>
    <section
      class="rac-layout rac-layout-center"
    >
      Search
    </section>
    <footer
      class="rac-layout-footer"
    >
      Footer
    </footer>
  </section>
</div>
`;
